client/allocrunner/taskrunner/task_runner.go

```diff
@@ -203,6 +203,8 @@ type TaskRunner struct {
 	// closed.
 	waitOnServers bool
 
+	restored bool
```

Comment:
Since this is only set in Restore() we don't need to worry about synchronizing writes but lets be optimistic about reads in-case another hook needs to read from this. We typically like to synchronize most reads from hooks on the taskrunner state just because of the concurrent nature of hooks.

Example:
https://github.com/hashicorp/nomad/blob/f02c163532238bb72baa3cd0df36e4143a3c909c/client/allocrunner/taskrunner/task_runner_getters.go#L97-L102

Reply:
I have a question about Restore(). Can be situations when task in restored state(in this PR we doesn't reset this fact - just doesn;t find place where we can do this), but resatrt of task may be called multiple times?